pkg/streamingpromql/types/ring_buffer_test.go

```diff
@@ -162,6 +162,52 @@ func testDiscardPointsBeforeThroughWrapAround[T any](t *testing.T, buf ringBuffe
 	shouldHavePoints(t, buf, points[5])
 }
 
+func TestRemoveLastPoint(t *testing.T) {
+	points := []promql.HPoint{
+		{T: 1, H: &histogram.FloatHistogram{Count: 100}},
+		{T: 2, H: &histogram.FloatHistogram{Count: 200}},
+		{T: 3, H: &histogram.FloatHistogram{Count: 300}},
+		{T: 4, H: &histogram.FloatHistogram{Count: 400}},
+	}
+
+	buf := &hPointRingBufferWrapper{&HPointRingBuffer{pool: &hPointPoolForRingBufferTesting{}}}
```

Comment:
Given this test only tests `HPointRingBuffer`s, can we just use a `HPointRingBuffer` directly here?

Reply:
it's to keep the generics simple for things like `shouldHavePoints`

Reply:
Ah, I missed that, thanks.